building-with-ocean/compute-to-data/user-defined-parameters.md

```diff
@@ -0,0 +1,183 @@
+---
+description: >-
+  Learn how to define and use custom parameters while downloading assets or
+  using dataset in Compute-to-data environment
+---
+
+# User defined parameters
+
+### Overview
+
+Ocean Protocol allows dataset buyers to provide custom parameters that can be used to fetch the downloaded data in a specific format, download a different type of data or pass some additional input to the algorithms in the Compute-to-Data job.
+
+There 2 are types of parameters that asset publishers can support:
+
+* User defined parameters
+* Algorithm custom parameters
+
+### Publish a dataset that uses custom parameters
+
+The dataset publisher can support these parameters to allow filtering, querying on the published dataset. The additional parameters that facilitate this are called `User defined parameters`. The Provider combines the original asset URL and the entered parameter values into a new URL and then streams the response from the modified URL back to the buyer.
+
+#### Use case for user defined parameters
+
+For example, if the publisher has published an URL `https://example.com` which serves large size historical weather data from all over the world. If the publisher wants to allow buyers to filter the data based on location, type of data, etc., it is possible to do so using user defined parameters.
+
+Suppose the publisher defines the following 2 parameters:
+
+* `location`: A string indicating region code
+* `type`: A string indicating the type of weather data. It can be temperature/humidity/pressure.
+
+Suppose the buyer wants to download the temperature data in the region code `XYZ`. While downloading the data buyer enters the desired parameter values using ocean.py or ocean.js.
```

Comment:
```suggestion
Suppose the buyer wants to download the temperature data in the region code `XYZ`. While downloading the data, the buyer enters the desired parameter values using ocean.py or ocean.js.
```

Reply:
Can this be done with ocean.js?

Reply:
Not as of now. But, I think it will be soon supported in ocean.js as well.